memote/support/biomass.py

```diff
@@ -176,14 +180,56 @@ def find_direct_metabolites(model, reaction):
         that are taken up to be consumed by the biomass reaction only.
 
     """
-    transport_reactions = set(helpers.find_transport_reactions(model))
-    exchange_reactions = set(model.exchanges)
-    biomass_reactions = set(helpers.find_biomass_reaction(model))
-
-    combined_set = transport_reactions | exchange_reactions | biomass_reactions
+    biomass_rxns = set(helpers.find_biomass_reaction(model))
+    tra_bou_bio_rxns = helpers.find_tra_bou_bio_reactions(model, biomass_rxns)
     precursors = find_biomass_precursors(model, reaction)
+    tra_bou_bio_mets = [met for met in precursors if
+                        met.reactions.issubset(tra_bou_bio_rxns)]
+    rxns_of_interest = set([rxn for met in tra_bou_bio_mets
+                            for rxn in met.reactions
+                            if rxn not in biomass_rxns])
+
+    solution = model.optimize()
+    if solution.objective_value == 0:
+        raise ValueError("objective value is zero. Model will not grow.")
 
-    return [met for met in precursors if met.reactions.issubset(combined_set)]
+    tra_bou_bio_fluxes = solution.fluxes[get_ids(rxns_of_interest)]
+    flux_sum = pd.DataFrame(index=tra_bou_bio_mets, columns=["sum"], data=0).T
+
+    main_comp = helpers.find_compartment_id_in_model(model, 'c')
+    exc_space = helpers.find_compartment_id_in_model(model, 'e')
+    for met in tra_bou_bio_mets:
+        for rxn in met.reactions:
+            not_biomass_rxn = rxn not in biomass_rxns
+            extracellular = met.compartment == exc_space
+            # if reactant metabolite in "e"
+            if met in rxn.reactants and not_biomass_rxn and extracellular:
+                product_comps = set([p.compartment for p in rxn.products])
+                # e reactant, no product (outward flux)
```

Comment:
Just fyi, and I'm certainly also guilty of this, but great comments go a long way in making your code accessible to others and your future self. These are somewhat cryptic and perhaps only clear to you once you've fully understood the code, at which point they become redundant :D

Reply:
Okay, I've tried clarifying the code!